api/services/diff/diff.proto

```diff
@@ -0,0 +1,60 @@
+syntax = "proto3";
+
+package containerd.v1;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/timestamp.proto";
+import "github.com/containerd/containerd/api/types/mount/mount.proto";
+import "github.com/containerd/containerd/api/types/descriptor/descriptor.proto";
+
+// Diff service creates and applies diffs
+service Diff {
+	// Extract extracts the content associated with the provided
+	// digests onto the provided mounts.
+	rpc Extract(ExtractRequest) returns (ExtractResponse);
+
+	rpc Diff(DiffRequest) returns (DiffResponse);
+}
+
+message ExtractRequest {
+	// Diff is the descriptor of the diff to be extracted
+	containerd.v1.types.Descriptor diff = 1;
+
+	repeated containerd.v1.types.Mount mounts = 2;
+}
+
+message ExtractResponse {
+	// Extracted is the descriptor for the object which was extracted.
+	// If the input was a compressed blob then the result will be
+	// the descriptor for the uncompressed blob.
+	containerd.v1.types.Descriptor extracted = 1;
+}
+
+message DiffRequest {
+	repeated containerd.v1.types.Mount lower_mounts = 1;
```

Comment:
lower/upper can be confused with overlay. I know we went with A/B  in the other APIs, but these fields need more context for comparison. Don't have a good suggestion.